### PR TITLE
Split `input_transform` into `context_input_transform` and `label_input_transform`

### PR DESCRIPTION
**Pull request by abdulfatir**
*Description of changes:* This splits `input_transform` into `context_input_transform` and `label_input_transform`. Previously, `input_transform` was being used for both context and label during training which would lead to incorrect results where `prediction_length` > `context_length`.

TODO:

- [x] Update docstrings
- [x] Test the training script

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/chronos/chronos.py [hunk 1]
```diff
@@ -336,10 +390,7 @@ class ChronosPipeline:
 
     tokenizer: ChronosTokenizer
     model: ChronosModel
-
-    def __init__(self, tokenizer, model):
-        self.tokenizer = tokenizer
-        self.model = model
+    forecast_type: Literal["samples", "quantiles"] = "samples"
```
Comment (lostella) [inline src/chronos/chronos.py:393]: Unrelated?

File: src/chronos/chronos.py [hunk 2]
```diff
@@ -170,16 +188,51 @@ def input_transform(
...
+        self, context: torch.Tensor
+    ) -> Tuple[torch.Tensor, torch.Tensor, torch.Tensor]:
+        length = context.shape[-1]
+
+        if length > self.config.context_length:
+            context = context[..., -self.config.context_length :]
+
+        token_ids, attention_mask, scale = self._input_transform(context=context)
+
+        if self.config.use_eos_token and self.config.model_type == "seq2seq":
```
Comment (lostella) [inline src/chronos/chronos.py:214]: is the `and self.config.model_type == "seq2seq"` part intended?

Comment (abdulfatir) [reply src/chronos/chronos.py:214]: Yes, we add EOS to the context only for seq2seq model. For causal, it should be [context, label, EOS].

File: src/chronos/chronos.py (hunk 1 above)
Comment (abdulfatir) [reply src/chronos/chronos.py]: Removed.

File: src/chronos/chronos.py [hunk 3]
```diff
@@ -59,27 +59,19 @@ class ChronosTokenizer:
...
-    def input_transform(
+    def context_input_transform(
         self,
         context: torch.Tensor,
-        tokenizer_state: Any = None,
-    ) -> Tuple[torch.Tensor, torch.Tensor, Any]:
+    ) -> Tuple:
         """
-        Turn a batch of time series into token IDs, attention map, and scale.
+        Turn a batch of time series into token IDs, attention map, and tokenizer_state.
```
Comment (lostella) [inline src/chronos/chronos.py:67]: The docstring of the third return argument can be improved (can't comment on it directly). Instead of

> An object that will be passed to ``output_transform``

better

> An object that can be passed to ``label_input_transform`` and ``output_transform``

**Review by lostella — COMMENTED** [review]
I have a proposal for slightly shorter names. Don't hate me, names are hard:
- `context_input_transform` -> `encode_context`
- `label_input_transform` -> `encode_label`
- `output_transform` -> `decode_samples`

What do you think? Of course if we change names then docstrings are to be updated

File: src/chronos/chronos.py (hunk 3 above)
Comment (abdulfatir) [reply src/chronos/chronos.py:67]: Fixed.

**Review by lostella — APPROVED** [review]
